app/controllers/vms_controller.rb

```diff
@@ -179,29 +156,30 @@ def filter_vm_categories(user)
   end
 
   def authorize_vm_access
-    @vm = VSphere::VirtualMachine.find_by_name params[:id]
-    return unless @vm
-
-    redirect_to vms_path unless current_user.admin? || @vm.users.include?(current_user)
-  end
+    configuration = VirtualMachineConfig.find_by_name params[:id]
 
-  def info_params
-    params.require(:vm_info).permit(sudo_user_ids: [], non_sudo_user_ids: [])
-  end
-
-  def config_params
-    params.require(:vm_info).permit(:description, :ip, :dns)
+    redirect_to vms_path unless current_user.admin? || (configuration&.all_users&.include?(current_user))
   end
 
   def authenticate_root_user_or_admin
-    @vm = VSphere::VirtualMachine.find_by_name(params[:id])
-    return unless @vm
+    configuration = VirtualMachineConfig.find_by_name params[:id]
 
-    redirect_to vms_path unless current_user.admin? || @vm.sudo_users.include?(current_user)
+    redirect_to vms_path unless current_user.admin? ||
+                                configuration && (configuration.sudo_users.include?(current_user) || configuration.responsible_users.include?(current_user))
   end
 
   def not_enough_resources(exception)
     redirect_back(fallback_location: root_path)
     flash[:alert] = exception.message
   end
+
+  def assign_vm
+    @vm = VSphere::VirtualMachine.find_by_name(params[:id])
+    render(template: 'errors/not_found', status: :not_found) if @vm.nil?
```

Comment:
Should we also `redirect_back fallback_location: dashboard_path ...` here, instead of rendering the (pretty empty) error page?

Reply:
As this code is also used in other places, maybe we should discuss this in general

Reply:
I had the same thought but then decided to use this methode because it was already in use elsewhere.

Using a redirect_back may however be advantageous on actions like update (POST, PATCH, etc.), in which a rendered page is not expected at all.
In these cases a redirect is a much better solution in my opinion

I would however want to use the vms_path as a fallback location instead of the dashboard_path as I think it is more strongly related with the vms controller.